docs/binary-data-format.rst

```diff
@@ -0,0 +1,63 @@
+.. _binary-data-format:
+
+Binary Data Format
+==================
+
+Futhark supports a binary format for input files, which this page
```

Comment:
I would prefer an introduction more like:

Futhark programs compiled to an executable support both textual and binary input.  Both are read via standard input, and can be mixed, such that one argument to an entry point may be binary, and another may be textual.  The binary input format takes up significantly less space on disk, and can be read much faster than the textual format.  This chapter describes the binary input format and its current limitations.  The input formats (whether textual or binary) are not used for Futhark programs compiled to libraries, which instead use whichever format is supported by their host language.

(Then the "currently reading..." stuff about limitations.)